src/cmd/ksh93/tests/leaks.sh

```diff
@@ -41,52 +40,85 @@ function test_reset
 	done
 }
 
-n=1000
 
-# one round to get to steady state -- sensitive to -x
+typeset -A foo
+typeset -lui i=0
 
-test_reset $n
-a=0$(vmstate --format='+%(size)u')
-b=0$(vmstate --format='+%(size)u')
+init_vsz=$(ps -o vsz $$ | tail -n 1 | tr -d '\n')
 
-test_reset $n
-a=0$(vmstate --format='+%(size)u')
-test_reset $n
-b=0$(vmstate --format='+%(size)u')
-
-if	(( b > a ))
-then	err_exit "variable value reset memory leak -- $((b-a)) bytes after $n iterations"
-fi
+for (( i=0; i<50000; i++ ))
+do
+	unset foo[bar]
+	typeset -A foo[bar]
+	foo[bar][elem0]="data0"
+	foo[bar][elem1]="data1"
+	foo[bar][elem2]="data2"
+	foo[bar][elem3]="data3"
+	foo[bar][elem4]="data4"
+	foo[bar][elem5]="data5"
+	foo[bar][elem6]="data6"
+	foo[bar][elem7]="data7"
+	foo[bar][elem8]="data8"
+	foo[bar][elem9]="data9"
+	foo[bar][elem9]="data9"
+done
 
-# buffer boundary tests
+curr_vsz=$(ps -o vsz $$ | tail -n 1 | tr -d '\n')
 
-for exp in 65535 65536
-do	got=$($SHELL -c 'x=$(printf "%.*c" '$exp' x); print ${#x}' 2>&1)
-	[[ $got == $exp ]] || err_exit "large command substitution failed -- expected $exp, got $got"
-done
+if [[ $init_vsz -lt $curr_vsz ]];
+then
+    err_exit "Memory leak on associate arrary"
+fi
 
-data="(v=;sid=;di=;hi=;ti='1328244300';lv='o';id='172.3.161.178';var=(k='conn_num._total';u=;fr=;l='Number of Connections';n='22';t='number';))"
-read -C stat <<< "$data"
-for ((i=0; i < 1; i++))
-do	print -r -- "$data"
-done |	while read -u$n -C stat
-	do	:
-	done	{n}<&0-
-a=0$(vmstate --format='+%(size)u')
-for ((i=0; i < 500; i++))
-do	print -r -- "$data"
-done |	while read -u$n -C stat
-	do	:
-	done	{n}<&0-
-b=0$(vmstate --format='+%(size)u')
-(( b > a )) && err_exit 'memory leak with read -C when deleting compound variable'
+#TODO: Enable these tests when vmstate is a builtin
 
-read -C stat <<< "$data"
-a=0$(vmstate --format='+%(size)u')
-for ((i=0; i < 500; i++))
-do      read -C stat <<< "$data"
-done
-b=0$(vmstate --format='+%(size)u')
-(( b > a )) && err_exit 'memory leak with read -C when using <<<'
+#builtin vmstate 2>/dev/null || exit 0
```

Comment:
Why does this test depend on the `vmstate` builtin if it is testing for memory leaks unrelated to that command? It should be possible to test for associative array leaks without resorting to unpredictable commands like `vmstate`. Including the preceding, not commented out, `ps -o vsz $$ | ...` pipeline. Regardless why not just remove this commented out code?

Reply:
Did you mean to remove ```+#builtin vmstate 2>/dev/null || exit 0``` ? I am fine with removing it.

These are legacy tests and I am not sure why they have choosen to use ```vmstate``` builtin. Using ```ps``` command instead of ```vmstate``` sounds good to me.